jmx_prometheus_javaagent/src/main/java/io/prometheus/jmx/JavaAgent.java

```diff
@@ -58,12 +61,15 @@ public static void premain(String agentArgument, Instrumentation instrumentation
             String host = config.host != null ? config.host : DEFAULT_HOST;
 
             httpServer =
-                    new HTTPServerFactory()
+                    HTTPServerFactory.getInstance()
```

Comment:
Should it be possible to use _only_ OpenTelemetry? I think the current implementation will open a Web server with a Prometheus scrape endpoint in any case, even if a user just wants OpenTelemetry.

Reply:
Correct.

The thought was that if there are issues, someone might want to hit the `/healthy/` or `/metrics` endpoint to check if the server is healthy/capturing metrics.

Also, the usage configuration currently requires the port...

```
java -javaagent:./jmx_prometheus_javaagent-1.0.1.jar=12345:config.yaml -jar yourJar.jar
```

```
java -jar jmx_prometheus_httpserver-1.0.1.jar 12345 config.yaml
```

I can see usage confusion if the port is specified but not being used (or the opposite... port not specified but required.) We don't look at the arguments before loading the configuration YAML.

Does it make sense to have a `jmx_prometheus_javaagent_opentelemetry.jar` or `jmx_opentelemetry_javagent.jar` project/artifact?

Reply:
I think it would be better to stick with one single JAR, having different deployment artifacts is confusing.

What do you think of making the port optional? So if you start with `-javaagent:./jmx_prometheus_javaagent-1.0.1.jar` without a port it will not expose a Prometheus endpoint?

Opening a network port may cause security concerns, and I assume there are users who want to avoid this when they switch to pushing OpenTelemetry.

Reply:
Sounds like a good approach. I'll make the change and update the documentation.